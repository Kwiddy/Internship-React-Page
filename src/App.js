import React from "react";
import logo from "./logo.svg";
import "./App.css";

function App() {
  return (
    <div>
      <div className="App">Herllsdfo, World!</div>

      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
      />
      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/icon?family=Material+Icons"
      />
    </div>
  );
}

export default App;
