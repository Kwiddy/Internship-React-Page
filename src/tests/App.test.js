import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { render, wait } from "@testing-library/react";
import { shallow, configure, mount } from "enzyme";
import renderer from "react-test-renderer";
import Button from "@material-ui/core/Button";
import DetTable from "../components/accountDetails.js";
import IconButton from "@material-ui/core/IconButton";
import sinon from "sinon";
import App from "../App";
import * as f from "../App";

configure({ adapter: new Adapter() });

/*
test("Renders header", () => {
  const { getByText } = render(<App />);
  const h1Element = getByText(/Lorem ipsum/i);
  const h2Element = getByText(/Dolor si amet/i);
  expect(h1Element).toBeInTheDocument();
  expect(h2Element).toBeInTheDocument();
});
*/
describe("Renders button-table buttons", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });
  it("should render cancelButton", () => {
    const mockFunction = jest.fn();
    const tree = shallow(<Button id="cancelButton" />);
    expect(tree).toMatchSnapshot();
  });
  it("should render updateButton correctly", () => {
    const tree = shallow(<Button id="updateButton" />);
    expect(tree).toMatchSnapshot();
  });
  //import * as f from "../App";
  //send eventhandler down to component as prop

  // Move button table into sep component
  // pass onclick as prop
  it("Test cancel button click event", () => {
    f.updateServices = jest.fn();
    const button = mount(
      <Button id="cancelButton" onClick={f.updateServices} />
    );
    expect(button.length).toBe(1);
    button.simulate("click");
    expect(f.updateServices).toHaveBeenCalled();
  });

  it("Test update button click event", () => {
    f.updateServices = jest.fn();
    const button = mount(
      <Button id="updateButton" onClick={f.updateServices} />
    );
    expect(button.length).toBe(1);
    button.simulate("click");
    expect(f.updateServices).toHaveBeenCalled();
  });
});

describe("App", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(<App />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
