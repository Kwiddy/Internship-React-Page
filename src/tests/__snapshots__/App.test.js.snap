// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="makeStyles-root-1"
  >
    <header>
      <div
        className="headText"
      >
        <h2
          className="MuiTypography-root MuiTypography-h1"
          fontSize="1"
          id="mainHead"
        >
          Your account
        </h2>
        <h2
          className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
          id="subHead"
        >
          Linked services
        </h2>
      </div>
    </header>
    <div
      className="page-wrapper"
    >
      <div
        className="page-block"
      >
        <div
          className="MuiCollapse-container MuiCollapse-hidden"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <div
                className="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
                id="action-alert"
                role="alert"
              >
                <div
                  className="MuiAlert-icon"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiAlert-message"
                >
                  Action completed successfully (See Console)
                </div>
                <div
                  className="MuiAlert-action"
                >
                  <button
                    aria-label="close"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                    disabled={false}
                    id="action-alert"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="account-details"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Account details
          </h5>
          <div>
            <table
              id="details"
            >
              <tbody>
                <tr>
                  <td
                    className="headCol"
                    style={
                      Object {
                        "width": "25%",
                      }
                    }
                  >
                    Account Name
                     
                  </td>
                  <td>
                    Guy Dale
                  </td>
                </tr>
                <tr>
                  <td
                    className="headCol"
                    style={
                      Object {
                        "width": "25%",
                      }
                    }
                  >
                    Account ID
                  </td>
                  <td>
                    AVLYM123
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="available-services"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Available services
          </h5>
          <div>
            <table
              id="services-table"
            >
              <tbody>
                <tr>
                  <th>
                    <input
                      onChange={[Function]}
                      type="checkbox"
                      value="checkedall"
                    />
                    Services
                  </th>
                  <th>
                     Description 
                  </th>
                </tr>
                <tr>
                  <td
                    className="service"
                  >
                    <li>
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                        value="Service 1"
                      />
                      Service 1
                    </li>
                  </td>
                  <td>
                    Description of service lorem ipsum dolor sit amet
                  </td>
                </tr>
                <tr>
                  <td
                    className="service"
                  >
                    <li>
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                        value="Service 2"
                      />
                      Service 2
                    </li>
                  </td>
                  <td>
                    Description of service lorem ipsum dolor sit amet
                  </td>
                </tr>
                <tr>
                  <td
                    className="service"
                  >
                    <li>
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                        value="Service 3"
                      />
                      Service 3
                    </li>
                  </td>
                  <td>
                    Description of service lorem ipsum dolor sit amet
                  </td>
                </tr>
                <tr>
                  <td
                    className="service"
                  >
                    <li>
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                        value="Service 4"
                      />
                      Service 4
                    </li>
                  </td>
                  <td>
                    Description of service lorem ipsum dolor sit amet
                  </td>
                </tr>
                <tr>
                  <td
                    className="service"
                  >
                    <li>
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                        value="Service 5"
                      />
                      Service 5
                    </li>
                  </td>
                  <td>
                    Description of service lorem ipsum dolor sit amet
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="button-table"
        >
          <table>
            <tbody>
              <tr>
                <td
                  id="leftCell"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-disableElevation"
                    disabled={false}
                    id="cancelButton"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Cancel
                    </span>
                  </button>
                </td>
                <td
                  id="rightCell"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
                    disabled={false}
                    id="updateButton"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Update services
                    </span>
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders button-table buttons should render cancelButton 1`] = `ShallowWrapper {}`;

exports[`Renders button-table buttons should render updateButton correctly 1`] = `ShallowWrapper {}`;
